SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverGameplayDebuggerContext.cpp

```diff
@@ -242,6 +255,84 @@ void USpatialNetDriverGameplayDebuggerContext::TickServer()
 
 		It.RemoveCurrent();
 	}
+
+	for (auto& Pair : TrackedEntities)
+	{
+		if (Pair.Value.Component.TrackPlayer)
+		{
+			if (!NetDriver->HasServerAuthority(Pair.Key))
+			{
+				// Do not track player if we don't have authority
+				continue;
+			}
+
+			TWeakObjectPtr<AGameplayDebuggerCategoryReplicator> ReplicatorWeakObjectPtr = Pair.Value.ReplicatorWeakObjectPtr;
+			AGameplayDebuggerCategoryReplicator* Replicator = ReplicatorWeakObjectPtr.Get();
+			if (Replicator == nullptr)
+			{
+				// Not unexpected - assume latency and wait for actor to appear
+				continue;
+			}
+
+			const AActor* PlayerControllerActor = Cast<AActor>(Replicator->GetReplicationOwner());
+			if (!ensureMsgf(PlayerControllerActor != nullptr, TEXT("Serious error: Cannot find player controller for replicated gameplay debugger actor")))
+			{
+				continue;
+			}
+
+			check(NetDriver->LoadBalanceStrategy);
+			auto GetActorVirtualWorkerId = [](const AActor& InActor, const USpatialNetDriver& InNetDriver, const SpatialGDK::FSubView& InSubView)->VirtualWorkerId
+			{
+				check(InNetDriver.PackageMap);
+
+				const Worker_EntityId EntityId = InNetDriver.PackageMap->GetEntityIdFromObject(&InActor);
+				if (!ensureMsgf(EntityId != SpatialConstants::INVALID_ENTITY_ID, TEXT("Serious error: Cannot find actor in package map")))
+				{
+					return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+				}
+
+				for (const SpatialGDK::ComponentData& Data : InSubView.GetView()[EntityId].Components)
+				{
+					if (Data.GetComponentId() == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
+					{
+						SpatialGDK::AuthorityIntent Intent = SpatialGDK::AuthorityIntent(Data.GetUnderlying());
+						const VirtualWorkerId VirtualWorkerId = Intent.VirtualWorkerId;
+						return VirtualWorkerId;
+					}
+				}
+
+				return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+			};
+
+			check(NetDriver && SubView);
+			const VirtualWorkerId PlayerControllerVirtualWorkerId = GetActorVirtualWorkerId(*PlayerControllerActor, *NetDriver, *SubView);
+			if (PlayerControllerVirtualWorkerId == SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+			{
+				continue;
+			}
+
+			const VirtualWorkerId ReplicatorVirtualWorkerId = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(*Replicator);
```

Comment:
I think this is my misunderstanding of how this works, but I assumed we'd look to update the target server if the current targeted server for PlayerControllerActor != the intended auth server for PlayerControllerActor. Whereas this is checking against the Replicator server != intended server for PlayerControllerActor.

Reply:
I think this is correct, but it is a weird indirection: Since this context created a wrapping LB strategy already, `WhoShouldHaveAuthority` will actually end up calling `GetActorDelegatedWorkerId` on this context object, which ends up returning the current assignment. I think it would be clearer if this simply called `GetActorDelegatedWorkerId`.